src/plot.js

```diff
@@ -59,11 +59,13 @@ export function plot(options = {}) {
   let facetChannels; // e.g. {fx: {value}, fy: {value}}
   let facetsIndex; // nested array of facet indexes [[0, 1, 3, …], [2, 5, …], …]
   let facetsExclude; // lazily-constructed opposite of facetsIndex
+  let facetDataLength;
   if (facet !== undefined) {
     const {x, y} = facet;
     if (x != null || y != null) {
       const facetData = arrayify(facet.data);
       if (facetData == null) throw new Error("missing facet data");
+      facetDataLength = facetData.length;
```

Comment:
Might as well expand the scope of the `facetData` variable instead of introducing a new `facetDataLength`.

Reply:
To be clear, you're just suggesting that we use `facetData.length`? Not sure why @Fil had added this variable.